tests/Unit/ControlSystem/Test_ControlSys.cpp

```diff
@@ -0,0 +1,45 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "ControlSystem/FunctionVsTime.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+SPECTRE_TEST_CASE("Unit.ControlSystem.FunctionVsTime",
+                  "[ControlSystem][Unit]") {
+  double t = 0;
+  // timestep must be an integer or power of 1/2,
+  // otherwise roundoff accumulates and CHECK will fail.
+  //
+  double dt = 0.8;
+  double FinalTime = 3.2;
+  const constexpr size_t DerivOrderFvt = 3;
+
+  // test two component system (x**3 and x**2)
+  std::vector<std::array<double, DerivOrderFvt + 1>> FuncInit{{0, 0, 0, 6},
+                                                              {0, 0, 2, 0}};
+  FunctionVsTimeNthDeriv<DerivOrderFvt> Fvt(t, FuncInit);
+
+  while (t < FinalTime) {
+    double value1 = t * t * t;
```

Comment:
inline these and use square(t) and cube(t)   defined in Utilities/ConstantExpressions.hpp